SparseGrids/TasmanianSparseGrid.cpp

```diff
@@ -818,6 +848,31 @@ template<typename FloatType> const FloatType* TasmanianSparseGrid::formCanonical
 template const float* TasmanianSparseGrid::formCanonicalPoints(const float *x, Data2D<float> &x_temp, int num_x) const;
 template const double* TasmanianSparseGrid::formCanonicalPoints<double>(const double *x, Data2D<double> &x_temp, int num_x) const;
 
+template<typename FloatType> void TasmanianSparseGrid::diffCanonicalTransform(const FloatType*, FloatType *jacobian_diag) const {
```

Comment:
* If the transform does not depend on `x`, then remove it. The first input is not needed.
* The way you're using this, you define a vector, initialize to zeros, then pass in the pointer, then reinitialize to 1.
    * declare the vector inside the method and return it

Reply:
Good points. I've fixed this in the latest PR.